x-pack/plugins/monitoring/server/es_client/instantiate_client.js

```diff
@@ -4,27 +4,32 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { bindKey, once } from 'lodash';
+import { once } from 'lodash';
 import { monitoringBulk } from '../kibana_monitoring/lib/monitoring_bulk';
-import { LOGGING_TAG } from '../../common/constants';
 
 /* Provide a dedicated Elasticsearch client for Monitoring
  * The connection options can be customized for the Monitoring application
  * This allows the app to connect to a dedicated monitoring cluster even if
  * Kibana itself is connected to a production cluster.
  */
 
-export function exposeClient({ elasticsearchConfig, events, log, elasticsearchPlugin }) {
+export function exposeClient({ elasticsearchConfig, log, elasticsearchPlugin }) {
   const isMonitoringCluster = hasMonitoringCluster(elasticsearchConfig);
+  // TODO: NP
+  if (elasticsearchConfig && elasticsearchConfig.hosts) {
```

Comment:
@restrry I'm confused by the need for this.

In [the elasticsearch config](https://github.com/elastic/kibana/blob/master/src/core/server/elasticsearch/elasticsearch_client_config.ts#L118), we check only for the existence of an array. Are we doing this kind of conversion somewhere in`src/core/server/elasticsearch` too?

Reply:
yes. raw elasticsearch config is wrapped in `ElasticsearchConfig` class with additional functionality. You might want to wrap your config as well https://github.com/elastic/kibana/blob/a08070ec1a75466f8913b0bfd0c5acea6ffb1226/src/core/server/elasticsearch/elasticsearch_config.ts#L261